lock-api/src/test/java/com/palantir/lock/watch/FailureCheckingLockWatchEventCacheTest.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.lock.watch;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.Mockito.when;
+
+import java.util.Optional;
+import java.util.UUID;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public final class FailureCheckingLockWatchEventCacheTest {
+
+    @Mock
+    private LockWatchEventCache defaultCache;
+    private LockWatchEventCache proxyCache;
+
+    @Before
+    public void before() {
+        proxyCache = FailureCheckingLockWatchEventCache.newProxyInstance(defaultCache);
+    }
+
+    @Test
+    public void failCausesNoOpCacheToBeUsed() {
+        RuntimeException runtimeException = new RuntimeException();
+        when(defaultCache.getCommitUpdate(anyLong())).thenThrow(runtimeException);
+        assertThatThrownBy(() -> proxyCache.getCommitUpdate(0L)).hasRootCause(runtimeException);
+
+        when(defaultCache.lastKnownVersion()).thenReturn(Optional.of(IdentifiedVersion.of(UUID.randomUUID(), 1L)));
+
+        // no op cache returns empty on last known version, so this should prove that we delegate there correctly
```

Comment:
You probably want to just pass the cache you want to fallback to into its constructor. It is weird to rely on this behavior for something you're testing, when there's a simple solution like I described.

Reply:
Done